README.md

```diff
@@ -88,220 +99,234 @@ kg_builder = SimpleKGPipeline(
-        -   New features
-        -   Bug fixes with high impact
-        -   Breaking changes
-    -   Minor changes:
-        -   Documentation improvements
-        -   Code refactoring without functional impact
-        -   Minor bug fixes
--   Keep `CHANGELOG.md` changes brief and focus on the most important changes.
+    the UI. Any other changes can be made in your fork and committed to the PR branch.
+- As you update your PR and apply changes, mark each conversation as [resolved](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#resolving-conversations).
+- Update the `CHANGELOG.md` if you have made significant changes to the project, these include:
+  - Major changes:
+    - New features
+    - Bug fixes with high impact
+    - Breaking changes
+  - Minor changes:
+    - Documentation improvements
+    - Code refactoring without functional impact
+    - Minor bug fixes
+- Keep `CHANGELOG.md` changes brief and focus on the most important changes.
 
 ### Updating the `CHANGELOG.md`
 
-1. When opening a PR, you can generate an edit suggestion by commenting on the GitHub PR [using CodiumAI](https://github.com/CodiumAI-Agent):
+1. You can automatically generate a changelog suggestion for your PR by commenting on it [using CodiumAI](https://github.com/CodiumAI-Agent):
 
 ```
 @CodiumAI-Agent /update_changelog
 ```
 
-2. Use this as a suggestion and update the `CHANGELOG.md` content under 'Next'.
+2. Edit the suggestion if necessary and update the appropriate subsection in the `CHANGELOG.md` file under 'Next'.
 3. Commit the changes.
 
-## Run tests
+## 🧪 Tests
 
-### Unit tests
+### Unit Tests
 
-This should run out of the box once the dependencies are installed.
+Install the project dependencies then run the following command to run the unit tests locally:
 
 ```bash
 poetry run pytest tests/unit
 ```
 
 ### E2E tests
 
-To run e2e tests you'd need to have some services running locally:
+To execute end-to-end (e2e) tests, you need the following services to be running locally:
 
--   neo4j
--   weaviate
--   weaviate-text2vec-transformers
+- neo4j
+- weaviate
+- weaviate-text2vec-transformers
 
-The easiest way to get it up and running is via Docker compose:
+The simplest way to set these up is by using Docker Compose:
 
 ```bash
 docker compose -f tests/e2e/docker-compose.yml up
 ```
 
-_(pro tip: if you suspect something in the databases are cached, run `docker compose -f tests/e2e/docker-compose.yml down` to remove them completely)_
+_(tip: If you encounter any caching issues within the databases, you can completely remove them by running `docker compose -f tests/e2e/docker-compose.yml down`)_
 
-Once the services are running, execute the following command to run the e2e tests.
+Once all the services are running, execute the following command to run the e2e tests:
 
 ```bash
 poetry run pytest tests/e2e
 ```
 
-## Further information
+## ℹ️ Additional Information
```

Comment:
Just wondering if we should mention ruff in the developer section?

Reply:
Good point. I'll add a little bit about pre-commit hooks too.